fastestimator/trace/trace.py

```diff
@@ -170,69 +177,35 @@ class EvalEssential(Trace):
-    def __init__(self, loss_keys: Set[str], monitor_names: Set[str]) -> None:
-        super().__init__(mode="eval",
-                         inputs=list(loss_keys) + list(monitor_names),
-                         outputs=self._configure_outputs(loss_keys, monitor_names))
+    def __init__(self, monitor_names: Set[str]) -> None:
+        super().__init__(mode="eval", inputs=monitor_names)
         self.eval_results = None
-        self.best_loss = None
-        self.since_best = 0
-
-    @staticmethod
-    def _configure_outputs(loss_keys: Set[str], monitor_names: Set[str]) -> List[str]:
-        """A function to determine the output keys of this Trace.
-
-        Args:
-            loss_keys: Which keys from the data dictionary correspond to loss values.
-            monitor_names: Any other keys which should be collected over the course of an eval epoch.
-
-        Returns:
-            A list of output keys. If there is exactly one `loss_key` then the system compute some extra outputs.
-        """
-        outputs = list(loss_keys) + list(monitor_names)
-        if len(loss_keys) == 1:
-            outputs.append("min_" + next(iter(loss_keys)))
-            outputs.append("since_best")
-        return outputs
 
     def on_epoch_begin(self, data: Data) -> None:
         self.eval_results = None
 
     def on_batch_end(self, data: Data) -> None:
         if self.eval_results is None:
-            self.eval_results = {k: [data[k]] for k in self.inputs}
+            self.eval_results = {key: [data[key]] for key in self.inputs if key in data}
         else:
             for key in self.inputs:
-                self.eval_results[key].append(data[key])
+                if key in data:
+                    self.eval_results[key].append(data[key])
 
     def on_epoch_end(self, data: Data) -> None:
         for key, value_list in self.eval_results.items():
             data.write_with_log(key, np.mean(np.array(value_list), axis=0))
-        if len(self.outputs) > len(self.inputs):  # There was exactly 1 loss key, so add the extra outputs
-            loss_name = self.inputs[0]
-            current_loss = data[loss_name]
-            if self.best_loss is None or current_loss < self.best_loss:
-                self.best_loss = current_loss
-                self.since_best = 0
-            else:
-                self.since_best += 1
-            data.write_with_log("min_" + loss_name, self.best_loss)
-            data.write_with_log("since_best", self.since_best)
 
 
 class Logger(Trace):
     """A Trace that prints log messages.
 
     Please don't add this trace into an estimator manually. FastEstimator will add it automatically.
-
-    Args:
-        extra_log_keys: A set of keys to print from the system buffer besides those it would normally print.
     """
-    def __init__(self, extra_log_keys: Set[str]) -> None:
-        super().__init__(inputs=extra_log_keys | {"*"})
+    def __init__(self) -> None:
+        super().__init__(inputs="*")
```

Comment:
inside the _print_message() function we can change this line now:
`for key, val in data.read_logs(to_set(self.inputs)).items():`
->
`for key, val in data.read_logs().items():`

Reply:
done

Reply:
the arg should actually be removed from the data class itself, not just here

Reply:
i made it optional, in case people want to read from second dictionary for whatever reason in the future.

Reply:
deleted the arg

Reply:
CSV logger needs update

Reply:
done